nncf/common/graph/graph.py

```diff
@@ -635,16 +640,11 @@ def _get_graph_for_visualization(self) -> nx.DiGraph:
         return out_graph
 
     def get_node_by_name(self, name: NNCFNodeName) -> NNCFNode:
-        matches = [node for node in self.get_all_nodes() if node.node_name == name]
-        if not matches:
+        node_id = self._node_name_to_node_id_map.get(name, None)
```

Comment:
The original ticket is about removing `get_all_nodes()` from NNCFGraph.  `get_all_nodes()` uses in many places such as algorithms, NNCFGraph and etc. Speeding up `get_all_nodes()` accelerates NNCF pipeline more than speeding up only `get_node_by_name`. My opinion that this PR is workaround for `get_node_by_name` because new entity was added to handle it and it does not fix the main problem.

Reply:
> Speeding up get_all_nodes() accelerates NNCF pipeline more than speeding up only get_node_by_name. 

I ran `cProfile` after my fix, and I got the following results.

| ncalls | tottime | percall | cumtime | percall | filename:lineno(function) |
|:------:|:---------:|:-------:|:----------:|:--------:|:----------------------------:|
|13      | 0.004     | 0.000   | 0.004      | 0.000    | nncf/nncf/common/graph/graph.py:271(get_all_nodes)|

According to this results, I don't consider that removing `get_all_nodes()` significantly improves execution time. The main problem was in the `get_node_by_name()` method.